doc/architecture/decisions/0008-add-first-class-citizen-fields-to-applications.md

```diff
@@ -0,0 +1,122 @@
+- update the UI code to send these additional fields to the API; or
+- use a [JsonLogic](https://jsonlogic.com) statement to fetch these items as part of the transformation process
+
+All of these approaches come with advantages and drawbacks:
+
+- saving the additional fields will require work at the API side, we would have to query the JSON object whenever it's updated to draw out the relevant fields. We would also have to consider what happens when the schema is updated
+- updating the UI code would further complicate the code at this level. We would have to update our page objects to pull out this data and marshal it into the correct format to send to the API, which would require extra conditionals and complexities. We also have no guarantee that the data would remain in sync.
+- using the JsonLogic statement in the transformer would require work to ensure this was updated every time the schema changed
+
+## Decision
+
+With all these approaches in mind, we have decided to use the transformers to
+add these additional fields to API responses using [JsonLogic](https://jsonlogic.com).
+
+Each schema entry in the database will also have a column in a JsonLogic
+format, mapping where to fetch the field from the JSON document. We can
+then process this query in the transformer. There are a couple of JsonLogic
+parsers we can use in a Kotlin application, such as [JSON Logic Java](https://github.com/jamsesso/json-logic-java),
+and [JSON Logic Kotlin](https://github.com/advantagefse/json-logic-kotlin).
+
+### Examples
+
+Assuming there is a `sentenceType` field in the JSON document
+that looks like this:
+
+```json
+{
+   "basic-information": {
+      "sentence-type": {
+         "sentenceType": "extendedDeterminate"
+      }
+   }
+}
+```
+
+Then there will be a `sentenceTypeQuery` field in the `application_schemas`
+table with the value:
+
+```json
+{ "var": "basic-information.sentence-type.sentenceType" }
+```
+
+If the schema changes, for example:
+
+```json
+{
+   "basic-info": {
+      "question1": {
+         "sentenceType": "extendedDeterminate"
+      }
+   }
+}
+```
+
+Then there will be a `sentenceTypeQuery` field in the new schema row with
+the value:
+
+```json
+{ "var": "basic-info.question1.sentenceType" }
+```
+
+There will be some situations where some first-class citizen fields (such
+as start date), may have to rely on multiple fields. In the case of start
+date, we will hav a `releaseDateQuery` field like this:
+
+```json
+{
+   "if":[
+      {
+         "==":[
+            {
+               "var":"basic-information.placement-date.startDateSameAsReleaseDate"
+            },
+            "yes"
+         ]
+      },
+      {
+         "var":"basic-information.release-date.releaseDate"
+      },
+      {
+         "var":"basic-information.placement-date.startDate"
+      }
+   ]
+}
+```
+
+This returns the value of `basic-information.release-date.releaseDate` if
```

Comment:
I find this examples incredibly helpful, thanks Stu 🙏
Using the first example: 
```json
{
   "basic-information": {
      "sentence-type": {
         "sentenceType": "extendedDeterminate"
      }
   }
}
```
What if instead of `sentence-type` changing, `sentenceType` changed? Would the schema update in the same way?